### PR TITLE
[FaceAPI] Add detection model argument on Detect and AddFace methods

### PR DESCRIPTION
**Pull request by TFR258**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by TFR258** [comment]
Spellchecker is failing due to mistake in specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/Encoding.json:1200:9 -> Unknown word (speicified)

File: specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json [hunk 1]
```diff
@@ -4238,6 +4268,24 @@
       "type": "boolean",
       "in": "query",
       "x-ms-parameter-location": "method"
+    },
+    "detectionModel": {
+      "name": "detectionModel",
+      "description": "The 'detectionModel' associated with the detected faceIds. Supported 'detectionModel' values include \"detection_01\" or \"detection_02\"",
```
Comment (lebronJ) [inline specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: Not only detected faceIds, but also the persisted faces in PG/LPG/FL/LFL.

Comment (lebronJ) [reply specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: Following the same style as the above `recognitionModel` should be good.

File: specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json [hunk 2]
```diff
@@ -1015,7 +1018,7 @@
     },
     "/detect": {
```
Comment (lebronJ) [inline specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: We should also apply same description to `/detect?overload=stream`.

**Review by lebronJ — APPROVED** [review]
Approved with suggestions.

File: specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json (hunk 1 above)
Comment (TFR258) [reply specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: Done =)

File: specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json (hunk 2 above)
Comment (TFR258) [reply specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: Done

**Comment by TFR258** [comment]
Hi @kpajdzik, @yangyuan  & @DavidLiCIG ,

There seems to have been an issue with the `Automation for azure-sdk-for-java` job. Can you restart it for me?

Can you also help reviewing this PR.

Thanks,
Tom

File: specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json [hunk 3]
```diff
@@ -4238,6 +4268,24 @@
       "type": "boolean",
       "in": "query",
       "x-ms-parameter-location": "method"
+    },
+    "detectionModel": {
+      "name": "detectionModel",
+      "description": "Name of detection model. Detection model' is used to detect faces in the submitted image. A detection model name can be provided when performing Face - Detect or (Large)FaceList - Create or (Large)PersonGroup - Create. The default value is 'detection_01', if another model is needed, please explicitly specify it.",
```
Comment (lebronJ) [inline specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: Add Face, rather than create facelists or persongroups.

**Review by lebronJ — CHANGES_REQUESTED** [review]

File: specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json [hunk 4]
```diff
@@ -967,7 +967,7 @@
     },
     "/persongroups/{personGroupId}/persons/{personId}/persistedfaces": {
```
Comment (lebronJ) [inline specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: One last comment, we should also unify the description for adding faces by `?overload=stream`.

File: specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json [hunk 5]
```diff
@@ -967,7 +967,7 @@
...
...ll cause failures.\n*   Adding/deleting faces to/from a same person will be processed sequentially. Adding/deleting faces to/from different persons are processed in parallel.\n* The minimum detectable face size is 36x36 pixels in an image no larger than 1920x1080 pixels. Images with dimensions higher than 1920x1080 pixels will need a proportionally larger minimum face size.\n* Different 'detectionModel' values can be provided. To use and compare different detection models, please refer to [How to specify a detection model](https://docs.microsoft.com/en-us/azure/cognitive-services/face/face-api-how-to-topics/specify-detection-model)\n  | Model | Recommended use-case(s) |\n  | ---------- | -------- |\n  | 'detection_01': | The default detection model for [PersonGroup Person - Add Face](/docs/services/563879b61984550e40cbbe8d/operations/563879b61984550f3039523b). Recommend for near frontal face detection. For scenarios with exceptionally large angle (head-pose) faces, occluded faces or wrong image orientation, the faces in such cases may not be detected. |\n  | 'detection_02': | Detection model released in 2019 May with improved accuracy especially on small, side and blurry faces. |",
```
Comment (lebronJ) [inline specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: Not `large`.

File: specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json [hunk 6]
```diff
@@ -2539,7 +2557,7 @@
...
```
Comment (lebronJ) [inline specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: Not `large`.

File: specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json (hunk 4 above)
Comment (lebronJ) [reply specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: Please remove all “LARGE” in the description.

File: specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json [hunk 7]
```diff
@@ -967,7 +967,7 @@
...
```
Comment (lebronJ) [inline specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: Please take care of all the `large` inside this description, they should be removed.

File: specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json [hunk 8]
```diff
@@ -2539,7 +2557,7 @@
...
```
Comment (lebronJ) [inline specification/cognitiveservices/data-plane/Face/stable/v1.0/Face.json]: Same.

**Review by lebronJ — CHANGES_REQUESTED** [review]

**Review by kpajdzik — APPROVED** [review]

**Review by lebronJ — APPROVED** [review]
LGTM.

**Comment by TFR258** [comment]
Hi @kpajdzik ,

This PR is ready to ship =)

The automation for `Automation for azure-sdk-for-java` failed. Can you help restarting it?

Thanks

**Comment by kpajdzik** [comment]
> Hi @kpajdzik ,
> 
> This PR is ready to ship =)
> 
> The automation for `Automation for azure-sdk-for-java` failed. Can you help restarting it?
> 
> Thanks

Java's generator has some problem with cognitive services Swagger. The owners still can generate the SDKs with manual adjustments so don't worry about it.

**Pull request merged**